frontend/pages/reports/label-generator.vue

```diff
@@ -1,315 +1,253 @@
+  aid = aid.toString();
+  let aidStr = aid.padStart(6, "0");
+  aidStr = aidStr.slice(0, 3) + "-" + aidStr.slice(3);
+  return aidStr;
+}
+
+function getQRCodeUrl(assetID: string): string {
+  let origin = displayProperties.baseURL.trim();
+  if (origin.endsWith("/")) {
+    origin = origin.slice(0, -1);
   }
+  const data = `${origin}/a/${assetID}`;
+  return route(`/qrcode`, { data: encodeURIComponent(data) });
+}
+
+function getItem(n: number, item: { name: string; location: { name: string } } | null): LabelData {
+  const assetID = fmtAssetID(n);
+  return {
+    url: getQRCodeUrl(assetID),
+    assetID,
+    name: item?.name ?? "_______________",
+    location: item?.location?.name ?? "_______________",
+  };
+}
 
-  function getQRCodeUrl(assetID: string): string {
-    let origin = displayProperties.baseURL.trim();
-
-    // remove trailing slash
-    if (origin.endsWith("/")) {
-      origin = origin.slice(0, -1);
-    }
-
-    const data = `${origin}/a/${assetID}`;
-
-    return route(`/qrcode`, { data: encodeURIComponent(data) });
+const { data: allFields } = await useAsyncData(async () => {
+  const { data, error } = await api.items.getAll();
+  if (error) {
+    return { items: [] };
   }
+  return data;
+});
 
-  function getItem(n: number, item: { name: string; location: { name: string } } | null): LabelData {
-    // format n into - seperated string with leading zeros
-
-    const assetID = fmtAssetID(n);
-
-    return {
-      url: getQRCodeUrl(assetID),
-      assetID,
-      name: item?.name ?? "_______________",
-      location: item?.location?.name ?? "_______________",
-    };
+const items = computed(() => {
+  if (displayProperties.assetRange > displayProperties.assetRangeMax) {
+    return [];
   }
 
-  const { data: allFields } = await useAsyncData(async () => {
-    const { data, error } = await api.items.getAll();
-
-    if (error) {
-      return {
-        items: [],
-      };
-    }
-
-    return data;
-  });
-
-  const items = computed(() => {
-    if (displayProperties.assetRange > displayProperties.assetRangeMax) {
-      return [];
-    }
-
-    const diff = displayProperties.assetRangeMax - displayProperties.assetRange;
-
-    if (diff > 999) {
-      return [];
-    }
+  const diff = displayProperties.assetRangeMax - displayProperties.assetRange;
+  if (diff > 999) {
+    return [];
+  }
 
-    const items: LabelData[] = [];
-    for (let i = displayProperties.assetRange; i < displayProperties.assetRangeMax; i++) {
-      const item = allFields?.value?.items?.[i];
-      if (item?.location) {
-        items.push(getItem(i, item as { location: { name: string }; name: string }));
-      } else {
-        items.push(getItem(i, null));
-      }
-    }
-    return items;
+  const items: LabelData[] = [];
+  for (let i = displayProperties.assetRange; i < displayProperties.assetRangeMax; i++) {
+    const item = allFields?.value?.items?.[i];
+    items.push(getItem(i, item?.location ? item : null));
```

Comment:
This is where the asset ID mismatch happens.  api.items.getAll() which is used to fill allFields appears to return the items in alphabetical order based on name (seems like the entire database, which may be inefficient if you have a large number of items and only want to print a few labels).  Loop index "i" is not associated with the actual assetID of allFields?.value?.items?.[i], it's just whatever order the array happened to get filled in.  This loop index i is what is used to create the QR code and assetID label in "getItem".  Even if the array did happen to get filled in the same order of assetID, this wouldn't handle a case where there was a gap in assetIDs.  

Unfortunately it doesn't look like api.items.getAll() returns an assetID.  Probably the right fix is changing that so it does, or you can look up the assetID based on the id property which is what I did in my comment on issue #289 but that seems more like a band-aid.